cloudevents/go/cde/README.md

```diff
@@ -0,0 +1,101 @@
+
+The idea behind this framework and command-line interface is to empower you  emitting events related to a Continuous Delivery to enable interoperability between different tools, measure performance and gain visibility of your CD processes.
+
+
+## Usage
+
+Download the binary `cde` or build from source, by clonning this repository and run `make build`
+
+Set `CDE_SINK` environment variable to define where CloudEvents will be sent. You can do this by running:
+
+`export CDE_SINK=http://my-cloudevent-broker` 
+
+You can use [SockEye](https://github.com/n3wscott/sockeye) to visualize the cloud events.
+
+Integrate with your existing pipelines, repositories, and environments.
+
+## Supported Events
+
+Use `cde --help`
+
+The following events are supported:
+- [Environment Events](#environment-events)
+- [Service Events](#service-events)
+- [PipelineRun Events](#pipelinerun-events)
+- [TaskRun Events](#taskrun-events)
+- [Repository Events](#repository-events)
+            
+
+# CDE Events
+
+The following events are currently supported: 
+
+## Environment Events
+- Properties
+    - Id
+    - Name
+    - URL
+      
+Examples: 
+
+- **cd.environment.created.v1** `./cde env created --id staging --name "Staging Environment" --repo http://github.com/user/myrepo --data event=data`
+- **cd.environment.modified.v1** `./cde env modified --id staging --name "Staging Environment" --repo http://github.com/user/myrepo --data event=data`
+- **cd.environment.deleted.v1** `./cde env deleted --id staging --name "Staging Environment" --repo http://github.com/user/myrepo --data event=data`
+
+
+## Service Events
+- Properties
+    - Name
+    - Version
+    - Environment Id
+
+Examples:
+
+- **cd.service.deployed.v1** `./cde service deployed --envId staging --name my-service --version 1.0.2 --data service=data`
+- **cd.service.upgraded.v1** `./cde service upgraded --envId staging --name my-service --version 1.0.3 --data service=data`
+- **cd.service.rolledback.v1** `./cde service upgraded --envId staging --name my-service --version 1.0.2 --data service=data`
+- **cd.service.removed.v1** `./cde service upgraded --envId staging --name my-service --version 1.0.2 --data service=data`
+
+## PipelineRun Events
+- Properties
+  - Id
+  - Name
+  - URL
+  - Status  
+  - Errors
+    
+Examples:
+- **cd.pipelinerun.started.v1**  `./cde pipelinerun queued --help --id pipe1 --name "My Pipeline" --status "Starting" --url "http://my-pipelinerunner" --errors "Hopfully no errors" --data pipeline=data`
+- **cd.pipelinerun.finished.v1** `./cde pipelinerun started --help --id pipe1 --name "My Pipeline" --status "Starting" --url "http://my-pipelinerunner" --errors "Hopfully no errors" --data pipeline=data`
+- **cd.pipelinerun.queued.v1**   `./cde pipelinerun finished --help --id pipe1 --name "My Pipeline" --status "Starting" --url "http://my-pipelinerunner" --errors "Hopfully no errors" --data pipeline=data`
+
+## TaskRun Events
```

Comment:
PipelineRun and TaskRun are very Tekton specific.
Maybe we could go with Pipeline (instead of PipelineRun) and PIpelineStage(Instead of TaskRun) ?
a PipelineStage would be part of the Pipeline.

Reply:
@waveywaves check my previous comment, does it make sense?

Reply:
it does now, thank for clarification